Source/WebCore/bindings/js/JSDOMConvertPromise.h

```diff
@@ -89,4 +91,25 @@ template<typename T> struct JSConverter<IDLPromise<T>> {
     }
 };
 
+template<typename T> struct JSConverter<IDLPromiseIgnoringSuspension<T>> {
+    static constexpr bool needsState = true;
+    static constexpr bool needsGlobalObject = true;
+
+    static JSC::JSValue convert(JSC::JSGlobalObject&, JSDOMGlobalObject&, DOMPromise& promise)
+    {
+        return promise.guardedObject();
```

Comment:
It’s a little subtle that `guardedObject` is effectively `promiseIgnoringSuspension`.